google/api/resource.proto

```diff
@@ -71,9 +100,41 @@ extend google.protobuf.FieldOptions {
   //       }];
   //     }
   //
-  // Only one of {`resource`, `resource_reference`} may be set.
+  // Only one of {`resource`, `resource_set`, `resource_reference`} may be set.
```

Comment:
There is a corner-case where both `resource` and `resource_set` can be set together.
It's when a resource is initially just a Resource (ie not a set), then it needs changing to a ResourceSet. To maintain backward compatibility the `resource` should remain, and the `resource_set` added.
We may wish to ignore this case for now, especially as resource-sets are still up for discussion.

Reply:
> I suspect I'm missing something obvious! But...
> This PR seems to remove quite a bit of stuff that is required. E.g. LRO `OperationData`, method-signature, field_behavor, etc...

This PR merges in `master`, and therefore incorporates #31.

Reply:
> We may wish to ignore this case for now, especially as resource-sets are still up for discussion.

Yeah. Comments are easy to fix.